### PR TITLE
utility that creates a `requirements.txt` file from `settings.ini`

### PR DESCRIPTION
**Pull request by hamelsmu**
This is useful for things like creating HF spaces

### CODE REVIEW

File: nbdev/release.py [hunk 1]
```diff
@@ -236,6 +236,15 @@ def write_conda_meta(path='conda'):
     _write_yaml(path, *_get_conda_meta())
 
 # %% ../nbs/api/release.ipynb 43
+def write_requirements(directory=None):
+    "Writes a `requirements.txt` file to `directory` based on settings.ini."
+    cfg = get_config()
+    d = cfg.config_path if not directory else Path(directory)
```
Comment (jph00) [inline nbdev/release.py:242]: Extremely minor, but you don't need to `not` there

```suggestion
    d = Path(directory) if directory else cfg.config_path
```

File: nbdev/release.py [hunk 2]
```diff
@@ -236,6 +236,15 @@ def write_conda_meta(path='conda'):
...
+    d.mkdir(exist_ok=True, parents=True)
```
Comment (jph00) [inline nbdev/release.py:243]: If you use fastcore's `Path.mk_write` patch, you could avoid this line.

**Comment by jph00** [comment]
Yay so happy to see this code @hamelsmu ! :D

**Comment by hamelsmu** [comment]
@jph00 Thanks for the careful code review, appreciate it

**Pull request merged**